### PR TITLE
Python packaging

### LINKED ISSUE #9: Assignment 2.2
- Create an issue and branch for this assignment.

- Make your code packageable and installable as a library

- Setup a github workflow job for building the package that does the following
  1. Install tree cli tool
  2. Display the directory tree in the workflow before building.
  3. Build the package
  4. Display the directory tree in the workflow after building.
  5. Upload the build artifacts using upload actions
 
- Setup a github workflow job for installing and  deploying (running) the nonstandardcode.py using the following steps:
  1. Conda setup from previous assignment (2.1)
  2. Download the artifact using download action 
  3. Install the package
  4. Run the nonstandardcode.py from the installed package.

### PR DESCRIPTION
**Pull request by Dhikshita-macherla**
Hence closes #9 
1. Created an issue and a new branch(enh/9/pypackaging)
2. Made the code packageable and installable as a library 
3. Workflow1 (Building_pkgs)
-   Installed tree cli tool
-  Built the package
- Uploaded the build artifacts
4. Workflow2(Install and Deploy)
- Downloaded the artifact using download action
- Installed the package
-  Executed nonstandardcode.py from installed package 

Screenshot of tree structure
Tree before build
image

Tree after build
image

2 jobs
image

Build artifact
image

### CODE REVIEW

File: .github/workflows/build-pkgs.yml [hunk 1]
```diff
@@ -0,0 +1,84 @@
+name: build-pkgs
+
+on:
+  pull_request:
+    branches:
+      - main
+  push:
+    branches:
+      - enh/9/pypackaging
```
Comment (abilashravi-ta) [inline .github/workflows/build-pkgs.yml:9]: Have just one trigger. If you have 2 overlapping triggers then the pipeline will run twice.
Also suppress the trigger in your previous pipeline (`ci-setup.yml`) whenever you add a new pipeline.

image

File: pyproject.toml [hunk 2]
```diff
@@ -0,0 +1,6 @@
+[build-system]
+requires = ["setuptools"]
+build-backend = "setuptools.build_meta"
+[project]
+name = "mypackage"
+version = "0.0.1"
```
Comment (abilashravi-ta) [inline pyproject.toml:6]: Give a meaningful name to the package instead of `mypackage`.

File: .github/workflows/ci-setup.yml [hunk 3]
```diff
@@ -33,8 +33,8 @@ jobs:
 
       - name: Run flake8 test
         run: |
-          python -m flake8 nonstandardcode.py
+          python -m flake8 src/mypackage/nonstandardcode.py
 
       - name: Run nonstandardcode.py
         run: |
-          python nonstandardcode.py
+          python src/mypackage/nonstandardcode.py
```
Comment (abilashravi-ta) [inline .github/workflows/ci-setup.yml:40]: No need for these changes. Just suppressing the trigger of this pipeline is enough.
As mentioned above, suppress the triggers in your previous pipeline whenever you add a new pipeline.

**Review by abilashravi-ta — CHANGES_REQUESTED** [review]
Don't commit `.tgz` file to the repo.
It will anyway get downloaded in the pipeline VM when we run the `nonstandardcode.py`.

File: .github/workflows/ci-setup.yml (hunk 3 above)
Comment (Dhikshita-macherla) [reply .github/workflows/ci-setup.yml]: I've changed all the corrections, Is it alright?

File: .github/workflows/build-pkgs.yml (hunk 1 above)
Comment (abilashravi-ta) [inline .github/workflows/build-pkgs.yml:9]: I still see 2 triggers. Use the `push` trigger only for your own testing purpose.
Have only the `pull_request` trigger in the final submission.

**Review by abilashravi-ta — CHANGES_REQUESTED** [review]

File: .github/workflows/build-pkgs.yml (hunk 1 above)
Comment (Dhikshita-macherla) [reply .github/workflows/build-pkgs.yml]: I've corrected it, Can you please take a look at it?

**Review by abilashravi-ta — APPROVED** [review]

**Pull request merged**